src/spi.rs

```diff
@@ -102,6 +102,9 @@ pub mod config {
     pub struct Config {
         pub baudrate: Hertz,
         pub data_mode: embedded_hal::spi::Mode,
+        pub no_dummy: bool,
+        pub dma_channel: u32,
```

Comment:
This is very unsafe in that the user has to consult the ESP-IDF documentation to figure out what to pass. I would prefer an `enum` here. Something like:
```
pub enum DMA {
    Disabled,
    Channel1,
    Channel2,
    Auto,
}
```

And then `From<DMA>` implementation for `spi_dma_chan_t`.